### PR TITLE
Resolve uefi guest fail to check device.map

### PR DESCRIPTION
**Pull request by weikunzz**
Because uefi guest has no device.map file

Signed-off-by: weikunzz <kuwei@redhat.com>

### CODE REVIEW

File: provider/v2v_vmcheck_helper.py [hunk 1]
```diff
@@ -212,11 +212,14 @@ def check_linux_vm(self):
 
         # 5. Check virtio disk partition
         logging.info("Checking virtio disk partition in device map")
-        if not self.checker.get_grub_device():
-            err_msg = "Not find vd? in disk partition"
-            self.log_err(err_msg)
+        if not self.checker.is_uefi_guest():
```
Comment (chunfuwen) [inline provider/v2v_vmcheck_helper.py]: line 215 and 216 merge into one line:
if not self.checker.is_uefi_guest() and not self.checker.get_grub_device():

File: provider/v2v_vmcheck_helper.py [hunk 2]
```diff
@@ -212,11 +212,14 @@ def check_linux_vm(self):
...
+            if not self.checker.get_grub_device():
+                err_msg = "Not find vd? in disk partition"
+                self.log_err(err_msg)
+            else:
```
Comment (chunfuwen) [inline provider/v2v_vmcheck_helper.py]: remove line 219 and 220

**Comment by weikunzz** [comment]
In uefi guest:
Sending command: ls /boot/efi/EFI/BOOT/
2018-01-10 14:56:55,144 client           L1078 DEBUG| Sending command: echo $?
2018-01-10 14:56:55,245 utils_v2v        L0257 DEBUG| Command return status: 0
2018-01-10 14:56:55,245 utils_v2v        L0258 DEBUG| Command output:
BOOTX64.EFI  fbx64.efi

2018-01-10 14:56:55,245 v2v_vmcheck_help L0222 INFO | The guest is uefi mode,skip the checkpoint

In not uefi guest:
 Checking virtio disk partition in device map
2018-01-10 14:25:56,693 client           L1078 DEBUG| Sending command: ls /boot/efi/EFI/BOOT/
2018-01-10 14:25:56,803 client           L1078 DEBUG| Sending command: echo $?
2018-01-10 14:25:56,906 utils_v2v        L0257 DEBUG| Command return status: 2
2018-01-10 14:25:56,906 utils_v2v        L0258 DEBUG| Command output:
ls: cannot access /boot/efi/EFI/BOOT/: No such file or directory

2018-01-10 14:25:56,906 client           L1078 DEBUG| Sending command: cat /boot/grub*/device.map
2018-01-10 14:25:57,019 client           L1078 DEBUG| Sending command: echo $?
2018-01-10 14:25:57,122 utils_v2v        L0257 DEBUG| Command return status: 0
2018-01-10 14:25:57,122 utils_v2v        L0258 DEBUG| Command output:
# this device map was generated by anaconda
(hd0)     /dev/vda

2018-01-10 14:25:57,122 v2v_vmcheck_help L0220 INFO | PASS

**Comment by chloerh** [comment]
:+1:

File: provider/v2v_vmcheck_helper.py (hunk 1 above)
Comment (weikunzz) [reply provider/v2v_vmcheck_helper.py]: Line 219 and 220 result is different from line 222 result. We should split the different result

**Comment by weikunzz** [comment]
The patch works together with https://github.com/avocado-framework/avocado-vt/pull/1292

File: provider/v2v_vmcheck_helper.py (hunk 1 above)
Comment (balamuruhans) [reply provider/v2v_vmcheck_helper.py]: agree with @chunfuwen

Comment (weikunzz) [reply provider/v2v_vmcheck_helper.py]: Hi,@balamuruhans line 219 and 220 must be retained.If line 215 and 216 merge into one line, how can i get two different result?

Comment (balamuruhans) [reply provider/v2v_vmcheck_helper.py]:
```python
if not self.checker.is_uefi_guest() and not self.checker.get_grub_device():
    err_msg = "Not find vd? in disk partition"
    self.log_err(err_msg)
elif self.checker.is_uefi_guest():
    logging.info("The guest is uefi mode,skip the checkpoint")
```

**Pull request closed without merging**